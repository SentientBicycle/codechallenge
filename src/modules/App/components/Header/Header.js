import React from 'react';
import styles from './Header.module.css';
import { Link } from 'react-router-dom';
import { CssConcat } from '../../../../utils/CssConcat';

export function Header(props, context) {

	const siteTitleCss = CssConcat(styles['vert-center'], styles['site-title']);
	const siteLinksCss = CssConcat(styles.link);
  return (
  	<nav className={styles['cg-nav']}>
	    <div className={styles.header}>
	      <div className={styles.content}>
	        <h1 className={siteTitleCss}>
	        	<Link to="/">
	        		Student Comparator
	        	</Link>
	        </h1>
	        <div className={siteLinksCss}>
	        	<Link to="/Students">
	        		Students
	        	</Link>
	        	<Link to="/Classes">
	        		Classes
	        	</Link>
	        </div>
	      </div>
	    </div>
	</nav>
  );
}


export default Header;
