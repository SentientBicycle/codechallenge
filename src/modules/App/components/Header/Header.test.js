import React from 'react';
import ReactDOM from 'react-dom';
import Header from './Header';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe("Header", () =>{
	it('Renders properly', () => {
	  const header = shallow(<Header/>);
	  expect(header).toMatchSnapshot();
	});
});