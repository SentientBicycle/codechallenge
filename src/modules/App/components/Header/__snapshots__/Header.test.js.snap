// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <div>
          <h1
            className=""
          >
            <Link
              replace={false}
              to="/"
            >
              Student Comparator
            </Link>
          </h1>
          <div
            className=""
          >
            <Link
              replace={false}
              to="/Students"
            />
          </div>
        </div>
      </div>,
      "className": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <h1
            className=""
          >
            <Link
              replace={false}
              to="/"
            >
              Student Comparator
            </Link>
          </h1>
          <div
            className=""
          >
            <Link
              replace={false}
              to="/Students"
            />
          </div>
        </div>,
        "className": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className=""
            >
              <Link
                replace={false}
                to="/"
              >
                Student Comparator
              </Link>
            </h1>,
            <div
              className=""
            >
              <Link
                replace={false}
                to="/Students"
              />
            </div>,
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/"
              >
                Student Comparator
              </Link>,
              "className": "",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Student Comparator",
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": "Student Comparator",
              "type": [Function],
            },
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/Students"
              />,
              "className": "",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "replace": false,
                "to": "/Students",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <div>
            <h1
              className=""
            >
              <Link
                replace={false}
                to="/"
              >
                Student Comparator
              </Link>
            </h1>
            <div
              className=""
            >
              <Link
                replace={false}
                to="/Students"
              />
            </div>
          </div>
        </div>,
        "className": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div>
            <h1
              className=""
            >
              <Link
                replace={false}
                to="/"
              >
                Student Comparator
              </Link>
            </h1>
            <div
              className=""
            >
              <Link
                replace={false}
                to="/Students"
              />
            </div>
          </div>,
          "className": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className=""
              >
                <Link
                  replace={false}
                  to="/"
                >
                  Student Comparator
                </Link>
              </h1>,
              <div
                className=""
              >
                <Link
                  replace={false}
                  to="/Students"
                />
              </div>,
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/"
                >
                  Student Comparator
                </Link>,
                "className": "",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Student Comparator",
                  "replace": false,
                  "to": "/",
                },
                "ref": null,
                "rendered": "Student Comparator",
                "type": [Function],
              },
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/Students"
                />,
                "className": "",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "replace": false,
                  "to": "/Students",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
